plugin/trino-session-property-managers/src/test/java/io/trino/plugin/session/TestSessionPropertyManagerInTransaction.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.session;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.containers.HiveHadoop;
+import io.trino.plugin.hive.containers.HiveMinioDataLake;
+import io.trino.plugin.hive.s3.S3HiveQueryRunner;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+import static io.trino.SystemSessionProperties.QUERY_MAX_EXECUTION_TIME;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+
+public class TestSessionPropertyManagerInTransaction
+        extends AbstractTestQueryFramework
+{
+    public static final File CONFIG_FILE = new File("src/test/resources/io/trino/plugin/session/file/session-property-config.properties");
+
+    private HiveMinioDataLake dockerizedS3DataLake;
+
+    @Override
+    protected QueryRunner createQueryRunner()
```

Comment:
can we use sth more lightweight than Hive here?

Reply:
I was not able to find anything that support transactions.
In memory, balckhole and mysql (which i tired) transactions are not allowed. Then I've check in code that only Hive have this unblocked.
Byt the way there is not information about that in documentation. I mean which connectors support and which not.

Reply:
I also had a hard time finding out in the documentation which connectors support transactions.

Reply:
From all I know no connector supports transaction and that is why it is not documented in SQL support section for the connectors ... at least thats what I remember from talking to @electrum about it

Reply:
@mosabua  From my finding and all the discussion around this PR Hive Connector does.

Reply:
Okay.. necessary follow up then - https://github.com/trinodb/trino/pull/10050 .. please review